### PR TITLE
Support cpufreq for t6031

### PR DESCRIPTION
**Pull request by dberlin**
This seems to work, and matches the info i can find.

Signed-off-by: Daniel Berlin <dberlin@dberlin.org>

### CODE REVIEW

File: src/cpufreq.c [hunk 1]
```diff
@@ -262,6 +280,8 @@ const struct feat_t *cpufreq_get_features(void)
         case T6021:
         case T6022:
             return t6020_features;
+        case T6031:
+            return t6031_features;
```
Comment (aruediger) [inline src/cpufreq.c:284]: same as T6021/T6022?

Comment (dberlin) [reply src/cpufreq.c:284]: For now, yes.  I assume it will end up slightly different in the end, which is why i didn't reuse the struct.

**Comment by marcan** [comment]
I'm not comfortable merging this until we at least confirm via macOS HV trace that it does the same thing. Getting this wrong has bit us *hard* in the past (M2 core lockups on deep sleep).

**Comment by dberlin** [comment]
That's fine.

**Comment by dberlin** [comment]
Note: I traced enough of the kext now to verify that this is at least as correct as the current one :)

**Pull request merged**